phedexClient.py

```diff
@@ -141,6 +141,43 @@ def getFileCountDataset(url, dataset):
     for block in result['phedex']['block']:
         files += block['files']
     return files
+
+def getFileCountPerBlock(url, dataset):
+    """
+    Returns the number of files per block in a dataset registered in phedex
+    """
+    result = phedexGet(url, '/phedex/datasvc/json/prod/blockreplicas?dataset='+dataset, auth=False)
+    if 'block' not in result['phedex']:
+        return {}
+    elif not result['phedex']['block']:
+        return {}
+    # we need blocks to be a list of tuples so we can create a set out of this
+    blocks = []
+    #check all blocks
+    for block in result['phedex']['block']:
+        # blocks.append({'name':block['name'],
+        #                'files':block['files']})
+        blocks.append((block['name'],block['files']))
+
+    return blocks
+
+
+def getFileNamesDataset(url, dataset):
```

Comment:
I don't see this function being called in checkor. You only use this function call for rucio client. @todor-ivanov are you sure we need this?

Reply:
That's is orreect. I was using this in the previous commit, but after redesigning the check to go on a block level it was not needed anymore, but I decided to leave it here anyway, since it could become useful. The utility function that is supposed to give the same result is doing it in parallel with a query to dbs which most of the times is not needed.